src/arkode/arkode_lsrkstep_impl.h

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------
+ * Programmer(s): Mustafa Aggul @ SMU
+ *---------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *---------------------------------------------------------------
+ * Implementation header file for ARKODE's LSRK time stepper
+ * module.
+ *--------------------------------------------------------------*/
+
+#ifndef _ARKODE_LSRKSTEP_IMPL_H
+#define _ARKODE_LSRKSTEP_IMPL_H
+
+#include <arkode/arkode_lsrkstep.h>
+
+#include "arkode_impl.h"
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+#define STAGE_MAX_LIMIT_DEFAULT 200
+#define DOM_EIG_SAFETY_DEFAULT  SUN_RCONST(1.01)
+#define DOM_EIG_FREQ_DEFAULT    25
+
+/*===============================================================
+  LSRK time step module data structure
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  Types : struct ARKodeLSRKStepMemRec, ARKodeLSRKStepMem
+  ---------------------------------------------------------------
+  The type ARKodeLSRKStepMem is type pointer to struct
+  ARKodeLSRKStepMemRec.  This structure contains fields to
+  perform an explicit Runge-Kutta time step.
+  ---------------------------------------------------------------*/
+typedef struct ARKodeLSRKStepMemRec
+{
+  /* LSRK problem specification */
+  ARKRhsFn fe;
+  ARKDomEigFn dom_eig_fn;
+
+  int q; /* method order               */
+  int p; /* embedding order            */
+
+  int req_stages; /* number of requested stages   */
+
+  ARKODE_LSRKMethodType LSRKmethod;
+
+  /* Counters and stats*/
+  long int nfe;               /* num fe calls       */
+  long int dom_eig_num_evals; /* num of dom_eig computations   */
+  int stage_max;              /* num of max stages used      */
+  int stage_max_limit;        /* max allowed num of stages     */
+  long int dom_eig_nst; /* num of step at which the last domainant eigenvalue was computed  */
+  long int step_nst; /* The number of successful steps. */
+
+  /* Spectral info */
+  sunrealtype lambdaR;         /* Real part of the dominated eigenvalue*/
+  sunrealtype lambdaI;         /* Imaginary part of the dominated eigenvalue*/
+  sunrealtype spectral_radius; /* spectral radius*/
+  sunrealtype spectral_radius_max; /* max spectral radius*/
+  sunrealtype spectral_radius_min; /* min spectral radius*/
+  sunrealtype dom_eig_safety; /* some safety factor for the user provided dom_eig*/
+  int dom_eig_freq; /* indicates dom_eig update after dom_eig_freq successful steps*/
```

Comment:
Update to `long int` to be consistent with the type used for similar options. The set function and docs will need to be updated accordingly. I'll add suggestions in those places in a sec.
```suggestion
  long int dom_eig_freq; /* indicates dom_eig update after dom_eig_freq successful steps*/
```